src/components/AddTripModal.js

```diff
@@ -0,0 +1,209 @@
+// helper functions
+import checkValidInput from '../utils/checkValidInput';
+// icons
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPenToSquare, faCircleMinus, faSquareCheck, faXmark } from '@fortawesome/free-solid-svg-icons';
+// db
+import firebase from '../firebase';
+import { getDatabase, ref, push } from 'firebase/database';
+// react
+import { useState } from 'react';
+// axios
+import axios from 'axios';
+
+function AddTripModal({ toggleAddModal }) {
+  // 'activity' state tracks the current activity being entered
+  // 'activities' state tracks the current entire array of activities to be added
+  // 'editedActivity' state tracks the current activity being edited
+
+  const [city, setCity] = useState('');
+  const [itinerary, setItinerary] = useState('');
+  const [activities, setActivities] = useState([]);
+  const [activity, setActivity] = useState('');
+  const [editedActivity, setEditedActivity] = useState('');
+  const [editIndex, setEditIndex] = useState('');
+
+  // profanity filter API
+  const hasProfanity = async (string) => {
+    const inputHasProfanity = await axios
+      .get('https://www.purgomalum.com/service/containsprofanity?text=' + string)
+      .then((response) => response.data)
+      .catch((error) => console.log('Profanity API Check Failed', error));
+    return inputHasProfanity;
+  };
+
```

Comment:
Great code. Can add a comment to outline the use of this variable for clarification